app/models/attribute.py

```diff
@@ -0,0 +1,139 @@
+from .. import db
+
+user_tag_associations_table = db.Table(
+    'user_tag_associations', db.Model.metadata,
+    db.Column('tag_id', db.Integer, db.ForeignKey('tags.id')),
+    db.Column('user_id', db.Integer, db.ForeignKey('users.id'))
+)
+
+resource_tag_associations_table = db.Table(
+    'resource_tag_associations', db.Model.metadata,
+    db.Column('tag_id', db.Integer, db.ForeignKey('tags.id')),
+    db.Column('resource_id', db.Integer, db.ForeignKey('resources.id'))
+)
+
+
+class Tag(db.Model):
+    __tablename__ = 'tags'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(30), unique=True)
+    users = db.relationship('User', secondary=user_tag_associations_table,
+                            backref='tags', lazy='dynamic')
+    resources = db.relationship('Resource',
+                                secondary=resource_tag_associations_table,
+                                backref='tags', lazy='dynamic')
+    type = db.Column(db.String(50))
+
+    __mapper_args__ = {
+        'polymorphic_identity': 'tag',
```

Comment:
do we need a `polymorphic_identity` for the parent? what happens if we remove the line?